Towerfall/Assets/_Main/Scripts/Player/PlayerManager.cs

```diff
@@ -7,35 +8,61 @@
 
 namespace Towerfall.Managers
 {
-    public partial class PlayerManager : IPlayerControllerInput
+    public partial class PlayerManager : IPlayerControllerInput, IPlayerControllerEvent
     {
         [Inject] private IPlayerInput _playerInput;
         [Inject] private IPlayerProperties _playerProperties;
 
-        private Subject<float> _jumpSubject = new Subject<float>();
-        private Subject<Vector2> _moveSubject = new Subject<Vector2>();
-        
-        public IObservable<float> Jump => _jumpSubject.AsObservable();
-        public IObservable<Vector2> Move => _moveSubject.AsObservable();
+        private Subject<float> _jumpStartSubject = new Subject<float>();
+        private Subject<float> _moveSubject = new Subject<float>();
+        private Subject<Vector2> _dashStartSubject = new Subject<Vector2>();
+
+        private Subject<Unit> _dashEndSubject = new Subject<Unit>();
         
-        private void PassJumpEvent()
+        public IObservable<float> JumpStart => _jumpStartSubject.AsObservable();
+        public IObservable<float> Move => _moveSubject.AsObservable();
+        public IObservable<Vector2> DashStart => _dashStartSubject.AsObservable();
+
+        public IObservable<Unit> DashEnd => _dashEndSubject.AsObservable();
+
+
+        private void PassJumpStartEvent()
         {
-            _jumpSubject.OnNext(_playerProperties.JumpForceMagnitude);
+            _jumpStartSubject.OnNext(_playerProperties.JumpForceMagnitude);
         }
 
-        private void PassMoveEvent(Vector2 moveDirection)
+        private void PassMoveEvent(float moveDirection)
         {
-            moveDirection.x *= _playerProperties.MoveAccelerationRate;
+            moveDirection *= _playerProperties.MoveAccelerationRate;
             _moveSubject.OnNext(moveDirection);
         }
+
+        private void PassDashStartEvent(Vector2 dashDirection)
+        {
+            Dash(dashDirection, _playerProperties.DashDuration).Forget();
+        }
     }
 
     public partial class PlayerManager : IInitializable
     {
         public void Initialize() 
         {
-            _playerInput.Jump.Subscribe(PassJumpEvent);
+            _playerInput.JumpStart.Subscribe(PassJumpStartEvent);
             _playerInput.Move.Subscribe(PassMoveEvent);
+            _playerInput.DashStart.Subscribe(PassDashStartEvent);
```

Comment:
Given how this is evolving I think the `IPlayerInput` observables should be named differently, maybe `Action`, so it would be `JumpStartAction`, this will make it less confusing since we are starting to use the same names here.

Reply:
d7dd0db43374ee04a601b14ef99a2961c364a854